File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long long card_number = get_long_long("Enter credit card number: ");

    int sum = 0;
    int alternate = 0;
    long long temp = card_number;

    while (temp > 0) {
        int digit = temp % 10;

        if (alternate) {
            digit *= 2;
            if (digit > 9) {
                digit -= 9;
            }
        }

        sum += digit;
        alternate = !alternate;
        temp /= 10;
    }

    if (sum % 10 == 0) {
        int length = 0;
        temp = card_number;
        while (temp > 0) {
            temp /= 10;
            length++;
        }

        long long divisor = 1;
        for (int i = 0; i < length - 2; i++) {
            divisor *= 10;
        }

        int first_two_digits = card_number / divisor;

        int first_digit = card_number / (divisor * 10);

        if ((first_two_digits >= 40 && first_two_digits <= 49) ||
            (first_digit == 4)) {
            printf("Visa\n");
        } else if (first_two_digits >= 51 && first_two_digits <= 55) {
            printf("MasterCard\n");
        } else if (first_two_digits == 34 || first_two_digits == 37) {
            printf("American Express\n");
        } else {
            printf("Unknown\n");
        }
    } else {
        printf("INVALID\n");
    }

    return 0;
}

File: credit/credit.py
from cs50 import get_int

cc=get_int("Number: ")
cc_str=str(cc)
sumval=0
for i in range(-2, -(len(cc_str)+1), -2):
    sumval += (2*int(cc_str[i]) %10) + (2*int(cc_str[i])//10)
for i in range(-1, -(len(cc_str)+1), -2):
    sumval += int(cc_str[i])
if sumval%10==0:
    first_digit=int(cc_str[0])
    first_two_digits=int(cc_str[:2])
    if first_digit==4 and len(cc_str) in [13,16]:
        print("VISA")
    elif first_two_digits in (51,56) and len(cc_str)==16:
        print("MASTERCARD")
    elif first_two_digits in [34,37] and len(cc_str)==15:
        print("AMEX")
    else:
        print("UNKNOWN")
else:
    print("INVALID")
